rotkehlchen/chain/ethereum/modules/polygon_pos_bridge/decoder.py

```diff
@@ -0,0 +1,155 @@
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.eth = A_ETH.resolve_to_crypto_asset()
+
+    def _decode_deposit(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes ETH and ERC20 bridge deposit events.
+
+        Ethereum to Polygon bridging uses a state sync mechanism to transfer data from ethereum
+        to polygon. See https://docs.polygon.technology/pos/architecture/bor/state-sync/
+
+        Mapped to POLYGON_STATE_SYNCER_ADDRESS. When it receives a STATE_SYNCED_TOPIC event,
+        a bridge deposit has been initiated and the corresponding spend event needs modified.
+        """
+        if context.tx_log.topics[0] != STATE_SYNCED_TOPIC:
+            return DEFAULT_DECODING_OUTPUT
+
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label is not None and
+                self.base.is_tracked(string_to_evm_address(event.location_label)) and
+                event.address in {BRIDGE_ADDRESS, ERC20_BRIDGE_ADDRESS, PLASMA_BRIDGE_ADDRESS}
+            ):
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.BRIDGE
+                event.notes = f'Bridge {event.balance.amount} {event.asset.resolve_to_asset_with_symbol().symbol} from Ethereum to Polygon PoS'  # noqa: E501
+                event.counterparty = CPT_POLYGON
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_withdraw(self, context: DecoderContext) -> DecodingOutput:
+        """Decode ETH and ERC20 bridge deposit events. (not Plasma, see _decode_plasma_withdraw)
+
+        Mapped to BRIDGE_ADDRESS, ETH_BRIDGE_ADDRESS, and ERC20_BRIDGE_ADDRESS. Also called
+        from _decode_plasma_process_exit. Checks if the topic indicates a withdrawal has been
+        initiated, and then modifies the corresponding receive event.
+        """
+        if context.tx_log.topics[0] not in {EXITED_ERC20_TOPIC, EXITED_ETHER_TOPIC, PROCESS_EXIT_TOPIC}:  # noqa: E501
+            return DEFAULT_DECODING_OUTPUT
+
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label is not None and
+                self.base.is_tracked(string_to_evm_address(event.location_label)) and
+                event.address in {BRIDGE_ADDRESS, ETH_BRIDGE_ADDRESS, ERC20_BRIDGE_ADDRESS, PLASMA_BRIDGE_ADDRESS}  # noqa: E501
+            ):
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.BRIDGE
+                event.notes = f'Bridge {event.balance.amount} {event.asset.resolve_to_asset_with_symbol().symbol} from Polygon PoS to Ethereum'  # noqa: E501
```

Comment:
we have a template for the notes related to bridging. You can use that function here if I'm not mistaken `bridge_match_transfer`